nameko_amqp_retry/rpc.py

```diff
@@ -0,0 +1,71 @@
+import sys
+
+from nameko.messaging import HeaderEncoder, QueueConsumer
+from nameko.rpc import Rpc as NamekoRpc
+from nameko.rpc import RpcConsumer as NamekoRpcConsumer
+
+from nameko_amqp_retry import Backoff, BackoffPublisher
+from nameko_amqp_retry.constants import (
+    CALL_ID_STACK_HEADER_KEY, RPC_METHOD_ID_HEADER_KEY)
+
+
+class RpcConsumer(NamekoRpcConsumer):
+
+    def handle_message(self, body, message):
+
+        # use the rpc_method_id if set, otherwise fall back to the routing key
+        method_id = message.headers.get(RPC_METHOD_ID_HEADER_KEY)
```

Comment:
That's exactly the reason. Actually RPC can be routed by header using _this extension_ only.